R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+#' Merge linelists from AFRO
+#'
+#' @param inputdirectory path to folder containing datasets
+#'
+#' @param outputdirectory path to folder where merged file is to be saved
+#'
+#' @param outputname character string to name merged file
+#'
+#' @param template logical (TRUE/FALSE) of whether the input data frame fits
+#' the standard template
+#'
+#' @importFrom rio import export
+#' @importFrom tidyr fill pivot_wider
+#' @importFrom matchmaker match_vec
+#' @importFrom dplyr bind_rows
+#' @importFrom janitor
+#' @author Alice Carr, Alex Spina
+#' @export
+
+#initialise list of ISO codes from the countries we are interested in merging
+#this is based off Francks files
+isotomerge<- c("BFA","CIV","COD","COG","DZA","GIN","KEN","LBR","MOZ","MUS","NAM","NER","RWA","SEN","SLE","STP","SWZ","SYC","TCD","UGA","ZWE")
+inputdirectory="inst/extdata/frank_linelists/"
+outputdirectory = tempdir()
+outputname = "Merged_linelist_"
+
+merge_linelist<- function(inputdirectory,
+                          outputdirectory = tempdir(),
+                          outputname = "Merged_linelist_", isotomerge) {
+
+  files <- base::list.files(path = inputdirectory, full.names = TRUE)
+
+  # create folder for output
+  base::dir.create(outputdirectory, showWarnings = FALSE)
+
+require(dplyr)
```

Comment:
dont require in function. put it in the imports above and then build when you are working on bits